pennylane/ops/op_math/controlled_decompositions.py

```diff
@@ -112,3 +211,208 @@ def decomp_circuit(op):
+
+    Not backpropagation compatible (as currently implemented). Use only with numpy.
+
+    This function decomposes a controlled single-qubit target operation using the
+    decomposition defined in section 3.2 of
+    `Vale et al. (2023) <https://arxiv.org/abs/2302.06377>`_.
+
+    .. warning:: This method will add a global phase for target operations that do not
+        belong to the SU(2) group.
+
+    Args:
+        u (np.ndarray): the target operation's matrix
+        target_wire (~.wires.Wires): the target wire of the controlled operation
+        control_wires (~.wires.Wires): the control wires of the operation
+
+    Returns:
+        list[Operation]: the decomposed operations
+    """
+    sx = qml.PauliX.compute_matrix()
+    sh = qml.Hadamard.compute_matrix()
+    sh_alt = sx @ sh @ sx
+
+    d, q = npl.eig(u)
+    d = np.diag(d)
+    q = _convert_to_real_diagonal(q)
+    b = _bisect_compute_b(q)
+    c1 = b @ sh_alt
+    c2t = b @ sh
+
+    mid = (len(control_wires) + 1) // 2  # for odd n, make lk bigger
+    lk = control_wires[:mid]
+    rk = control_wires[mid:]
+
+    op_mcx1 = qml.MultiControlledX(control_wires=lk, wires=target_wire, work_wires=rk)
+    op_mcx2 = qml.MultiControlledX(control_wires=rk, wires=target_wire, work_wires=lk)
+    op_c1 = qml.QubitUnitary(c1, target_wire)
+    op_c1t = qml.QubitUnitary(_matrix_adjoint(c1), target_wire)
+    op_c2t = qml.QubitUnitary(c2t, target_wire)
+    op_c2 = qml.QubitUnitary(_matrix_adjoint(c2t), target_wire)
+
+    inner = _ctrl_decomp_bisect_od(d, target_wire, control_wires)
+    result = [op_c2t, op_mcx2, op_c1t, op_mcx1] + inner + [op_mcx1, op_c1, op_mcx2, op_c2]
+    # cancel out adjacent op_mcx1
+    result = result[:3] + result[5:]
+    return result
+
+
+def ctrl_decomp_bisect(
+    target_operation: typing.Union[Operator, typing.Tuple[np.ndarray, Wires]],
+    control_wires: Wires,
+):
+    """Decompose the controlled version of a target single-qubit operation
+
+    Not backpropagation compatible (as currently implemented). Use only with numpy.
+
+    Automatically selects the best algorithm based on the matrix (uses specialized more efficient
+    algorithms if the matrix has a certain form, otherwise falls back to the general algorithm).
+
+    .. warning:: This method will add a global phase for target operations that do not
+        belong to the SU(2) group.
+
+    Args:
+        target_operation (~.operation.Operator): the target operation to decompose
+        control_wires (~.wires.Wires): the control wires of the operation
+
+    Returns:
+        list[Operation]: the decomposed operations
+
+    Raises:
+        ValueError: if ``target_operation`` is not a single-qubit operation
+
+    """
+    if isinstance(target_operation, Operator):
+        report_as = type(target_operation).__name__
+        target_operation = target_operation.matrix(), target_operation.wires
+    else:
+        report_as = f"anonymous operator with {target_operation[1]} wires"
```

Comment:
Since this decomposition is a helper for `Controlled.decomposition`,  I think this if-else should be removed and `target_operation` should only be `Operator`. The only use case for this that I can think of is when a controlled `QubitUnitary` is being decomposed, and even then the matrix and target wire are available through `target_operation.matrix` and `target_operation.wires`.